pkg/controller/dns/letsencrypt.go

```diff
@@ -0,0 +1,295 @@
+package dns
+
+import (
+	"context"
+	"crypto"
+	"crypto/ecdsa"
+	"crypto/elliptic"
+	"crypto/rand"
+	"crypto/x509"
+	"encoding/json"
+	"encoding/pem"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/acorn-io/acorn/pkg/labels"
+	"github.com/acorn-io/acorn/pkg/system"
+	"github.com/acorn-io/baaah/pkg/router"
+	"github.com/go-acme/lego/v4/certcrypto"
+	"github.com/go-acme/lego/v4/certificate"
+	"github.com/go-acme/lego/v4/lego"
+	"github.com/go-acme/lego/v4/registration"
+	"github.com/sirupsen/logrus"
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	kclient "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	LEEndpoint = "https://acme-staging-v02.api.letsencrypt.org/directory"
```

Comment:
This is going to have to come from config.  The only reason the existing DNS was a compile time option is we wanted to change the default value for dev builds. But for LE the default should always be staging and the user should opt into production. We should prompt during install on which one they want (along with email address for production, staging can use a dummy email).

Reply:
Alright, two things here
1. LE staging certs are not trusted by default in browsers.. that's not a problem? Could as well use self-signed certs then, right?
2. Is there some standard way defined already to ask for user input? Also, can one specify a config file upon installation?

Reply:
Oh wow... sorry for my still newborn-son-influenced brain... You probably just mean some new flags, right? :man_facepalming: